### PR TITLE
added Jenkinsfile

### PR DESCRIPTION
**Pull request by guineveresaenger**
For testing purposes

### CODE REVIEW

**Comment by coffeepac** [comment]
the container or customContainer call sets which container in the pod the enclosed commands will be run in.  so if you want the `fetch credentials` step to run on k2-tools you will need to have a `customContainer('k2-tools')` call before the `fetch credentials` step.

**Comment by joejulian** [comment]
Does the custom jnlp build stuff need to be in this? Seems like it doesn't.

**Comment by coffeepac** [comment]
It doesn't. But it should be in its own repo woth CI and not just shoved into k2.  I was hoping upstream would resolve quickly but thats not looking likely.  

Ill file an issue to split it out.  Should only take 2 hours for that one and the e2e-tester.

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+
+                // }
+
+                try {
+                    stage('Test: Cloud') {
+                        parallel ( 
+                            "aws": {
+                                kubesh "export helm_override_`echo ${JOB_BASE_NAME}-${BUILD_ID} " + '| tr \'[:upper:]\' \'[:lower:]\' | tr \'-\' \'_\'`=false' + " && go/src/github.com/samsung-cnct/k2cli/k2cli -vvv cluster up /var/lib/docker/scratch/k2cli-${env.JOB_BASE_NAME}-${env.BUILD_ID}/aws/config.yaml --output /var/lib/docker/scratch/k2cli-${env.JOB_BASE_NAME}-${env.BUILD_ID}/aws/"
+                            },
```
Comment (mikeln) [inline Jenkinsfile]: should not have "&&" or th   this starts a second line, and essentially a second shell.  Also, "export" should not be there as this is a temporary env setting.
Checkout the k2/Jenkinsfile.  Also here is the format correctly:

```kubesh "env helm_override_`echo ${JOB_BASE_NAME}-${BUILD_ID} " + '| tr \'[:upper:]\' \'[:lower:]\' | tr \'-\' \'_\'`=false <your commnand here>```

same in all these lines below.

File: build-scripts/golang-container/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+FROM golang:latest
+
```
Comment (mikeln) [inline build-scripts/golang-container/Dockerfile]: you don't want latest.   you want a specific version of golang. `golang:1.8.3`

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+
+                // }
+
+                try {
+                    stage('Test: Cloud') {
+                        parallel ( 
+                            "aws": {
+                                kubesh "env helm_override_`echo ${JOB_BASE_NAME}-${BUILD_ID} " + '| tr \'[:upper:]\' \'[:lower:]\' | tr \'-\' \'_\'`=false' + " go/src/github.com/samsung-cnct/k2cli/k2cli -vvv cluster up /var/lib/docker/scratch/k2cli-${env.JOB_BASE_NAME}-${env.BUILD_ID}/aws/config.yaml --output /var/lib/docker/scratch/k2cli-${env.JOB_BASE_NAME}-${env.BUILD_ID}/aws/"
+                            },
```
Comment (mikeln) [inline Jenkinsfile]: `helm_override_kca`

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+                            "gke": {
+                                kubesh "env helm_override_`echo ${JOB_BASE_NAME}-${BUILD_ID} " + '| tr \'[:upper:]\' \'[:lower:]\' | tr \'-\' \'_\'`=false' + " go/src/github.com/samsung-cnct/k2cli/k2cli -vvv cluster up /var/lib/docker/scratch/k2cli-${env.JOB_BASE_NAME}-${env.BUILD_ID}/gke/config.yaml --output /var/lib/docker/scratch/k2cli-${env.JOB_BASE_NAME}-${env.BUILD_ID}/gke/"
+                            }
```
Comment (mikeln) [inline Jenkinsfile]: `helm_override_kcg`

Comment (mikeln) [inline Jenkinsfile]: and the same on down... kca for aws, and kcg for gke.

**Comment by joejulian** [comment]
:tada:

**Pull request merged**